### DISCUSSION TITLE
What's the best solution for deploying to Azure?

### DISCUSSION THREAD

**Discussion by pokertournamentassistant** [OrchardCMS/OrchardCore#14030] [Q&A]
Hello,
i've deployed created a web site using orchard. It has a custom module and a theme. I would like to deploy it to azure but don't know what should be the best strategy.
- If i use a web app with code deployment (no container), i could connect to a SQL Database, but the connection string will be stored in App_Data Folder. I could use managed identities though but i don't know if Orchard Core is able to do it. On the other hand i could use SQL Authentication but how can i configure Orchard to use a connection string in a KeyVault?
- If i want to use SQL Lite, i will have all the data in the App_Data folder and it will not be easy to backup.
- It seems that containerised hosting would be a better solution. How can i map the App_Data to a storage.
Therefore what is the best approach? For single tenant and for multitenant. Is a SQL Lite DB created for each tenant?
Thanks a lot for your feedbacks.

**Comment by MikeAlhayek** [comment]
I like to use docker containers. This will help me roll out and roll back easily. Also, I can test out the containers before rolling container to production. You should not depend on `App_Data`. Instead you may want to use `OrchardCore.Media.Azure.Storage` to store any media so they are not stored in the `App_Data`. 

Also, you'll need to use `OrchardCore.DataProtection.Azure` to store the protection keys for each site into a blob storage. I would not use SQL Lite for production. If you use Azure SQL Database, you can use prefix or schema to create different tables for each tenents on the same SQL Server.

If you want to scale out your instance to more than 1 instance, you'll need to also enable caching like Redis cache. `OrchardCore.Redis`, `OrchardCore.Redis.Cache`, `OrchardCore.Redis.DataProtection`, `OrchardCore.Redis.Lock`.

I hope this helps

**Comment by ns8482e** [comment]
For azure you could deploy OrchardCore as app service, containers or Kubernetes(AKS)

Things to consider 

- Storing media on blob vs App_Data.
- Storing tenant configuration on blob vs App_Data.
- Storing content type and content definition in SQL Azure.
- Store content data in SQLAzure.
- Storing Data protection keys
